internal/conf/configuration.go

```diff
@@ -102,14 +102,29 @@ type JWTConfiguration struct {
 	KeyID            string   `json:"key_id" split_words:"true"`
 }
 
+type MFAFactorTypeConfiguration struct {
+	EnrollEnabled bool `split_words:"true" default:"true"`
+	VerifyEnabled bool `split_words:"true" default:"true"`
+}
+
+type SMSFactorTypeConfiguration struct {
+	MFAFactorTypeConfiguration
+	OtpLength    int                `json:"otp_length" split_words:"true"`
+	SMSTemplate  *template.Template `json:"-"`
+	MaxFrequency time.Duration      `json:"max_frequency" split_words:"true"`
+	Template     string             `json:"template"`
+}
+
 // MFAConfiguration holds all the MFA related Configuration
 type MFAConfiguration struct {
-	Enabled                     bool          `default:"false"`
-	ChallengeExpiryDuration     float64       `json:"challenge_expiry_duration" default:"300" split_words:"true"`
-	FactorExpiryDuration        time.Duration `json:"factor_expiry_duration" default:"300s" split_words:"true"`
-	RateLimitChallengeAndVerify float64       `split_words:"true" default:"15"`
-	MaxEnrolledFactors          float64       `split_words:"true" default:"10"`
-	MaxVerifiedFactors          int           `split_words:"true" default:"10"`
+	Enabled                     bool                       `default:"false"`
+	ChallengeExpiryDuration     float64                    `json:"challenge_expiry_duration" default:"300" split_words:"true"`
+	FactorExpiryDuration        time.Duration              `json:"factor_expiry_duration" default:"300s" split_words:"true"`
+	RateLimitChallengeAndVerify float64                    `split_words:"true" default:"15"`
+	MaxEnrolledFactors          float64                    `split_words:"true" default:"10"`
+	MaxVerifiedFactors          int                        `split_words:"true" default:"10"`
```

Comment:
why is `MaxEnrolledFactors` a float64 instead of int?
```suggestion
	MaxEnrolledFactors         int                    `split_words:"true" default:"10"`
	MaxVerifiedFactors          int                    `split_words:"true" default:"10"`
```

Reply:
Likely an error during the TOTP implementation which was left unchanged. 

Probably can leave a comment here to clarify this.